api/src/main/java/edu/cornell/mannlib/vitro/webapp/audit/controller/AuditController.java

```diff
@@ -0,0 +1,334 @@
+        // Pass the results to Freemarker
+        body.put("results", results);
+
+        // Create next / previous links
+        if (offset > 0) {
+            body.put("prevPage", getPreviousPageLink(offset, limit, sdf.format(startDate), sdf.format(endDate), order,
+                    userUri, graphUri, vreq.getServletPath()));
+        }
+        if (offset < (results.getTotal() - limit)) {
+            body.put("nextPage", getNextPageLink(offset, limit, sdf.format(startDate), sdf.format(endDate), order,
+                    userUri, graphUri, vreq.getServletPath()));
+        }
+
+        // Pass the user name to Freemarker
+        if (StringUtils.isNotEmpty(acc.getFirstName())) {
+            if (StringUtils.isNoneEmpty(acc.getLastName())) {
+                body.put("username", acc.getFirstName() + " " + acc.getLastName());
+            } else {
+                body.put("username", acc.getFirstName());
+            }
+        } else if (StringUtils.isNotEmpty(acc.getEmailAddress())) {
+            body.put("username", acc.getEmailAddress());
+        } else {
+            body.put("username", "");
+        }
+
+        // Pass the helper methods to Freemaker
+        body.put("listAddedStatements", new ListAddedStatementsMethod());
+        body.put("listRemovedStatements", new ListRemovedStatementsMethod());
+
+        // Return the default template and parameters
+        return new TemplateResponseValues(TEMPLATE_DEFAULT, body);
+    }
+
+    private boolean isAdmin(UserAccount acc) {
+        if (acc.isRootUser()) {
+            return true;
+        }
+        Set<String> roles = acc.getPermissionSetUris();
+        return (roles.contains(PermissionSets.URI_DBA));
+    }
+
+    private void setUserData(List<AuditChangeSet> list, UserAccountsDao uad) {
+        for (AuditChangeSet acs : list) {
+            UserAccount account = uad.getUserAccountByUri(acs.getUserId());
+            if (account == null) {
+                continue;
+            }
+            acs.setUserFirstName(account.getFirstName());
+            acs.setUserLastName(account.getLastName());
+            acs.setUserEmail(account.getEmailAddress());
+        }
+    }
+
+    /**
+     * Get the number of entries to show per page, or a default value of 10
+     *
+     * @param vreq
+     * @return
+     */
+    private int getLimit(VitroRequest vreq) {
+        int limit = 0;
+        try {
+            limit = Integer.parseInt(vreq.getParameter(PARAM_LIMIT));
+        } catch (Throwable e) {
+            log.debug(e, e);
+            limit = 10;
+        }
+        return limit;
+    }
+
+    /**
+     * Get the page offset, or a default of 0
+     *
+     * @param vreq
+     * @return
+     */
+    private int getOffset(VitroRequest vreq) {
+        int offset = 0;
+        try {
+            offset = Integer.parseInt(vreq.getParameter(PARAM_OFFSET));
+        } catch (Throwable e) {
+            log.debug(e, e);
+            offset = 0;
```

Comment:
This is a redundant reassignment of offset variable as if the exception was thrown it would not of reassigned in the try block.

Reply:
fixed